server/controllers/DataAccessLayer.js

```diff
@@ -56,12 +57,12 @@ exports.AddUserToFile = function(user) {
  */
 exports.UpdateUser = function(user) {
     // Find the index in the cache for the original User object
-    let index = cachedUsers.findIndex(x => x.username === user.username && x.email === user.email);
+    let index = cachedUsers.findIndex(x => x.id === user.id);
     // Get a pointer to the original User object
     let originalUser = cachedUsers[index];
 
     // Create a string that will be searched for within the text file
-    let originalUserString = originalUser.username + ";" + originalUser.password + ";" + 
+    let originalUserString = originalUser.id + ";" + originalUser.username + ";" + originalUser.password + ";" + 
```

Comment:
I found these semi-colons a little jarring to look at, would commas be a better fit (similar to a CSV)

Reply:
I've converted them to commas